gramineproject/gramine
# [WIP] Support project Amber services (SaaS)

## Description of the changes
This is a **WIP** PR for the purpose of feature experiment and feedback collection. Regarding the design proposal, please refer to the issue #822.
With this PR, The Gramine can directly talk with Project Amber alpha services, thus hiding the details of token retrieval workflow from user-space application and its dependencies. 
Please notice that this patch is based on Gramine v1.2, supports token retrieval from project Amber alpha service only. There are rooms for code improvement and some extra code might be identified for future use.

Fixes #822 

## How to test this PR?
There is a example which can be used to verify the basic functionality this PR provided if you can access Project Amber alpha services on your platform.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/gramine/828)
